src/dnssec/sign/keys/keyset.rs

```diff
@@ -1274,48 +1902,181 @@ fn zsk_roll(rollop: RollOp<'_>, ks: &mut KeySet) -> Result<(), Error> {
     Ok(())
 }
 
-fn zsk_roll_actions(rollstate: RollState) -> Vec<Action> {
-    let mut actions = Vec::new();
-    match rollstate {
-        RollState::Propagation1 => {
-            actions.push(Action::UpdateDnskeyRrset);
-            actions.push(Action::ReportDnskeyPropagated);
-        }
-        RollState::CacheExpire1(_) => (),
-        RollState::Propagation2 => {
-            actions.push(Action::UpdateRrsig);
-            actions.push(Action::ReportRrsigPropagated);
-        }
-        RollState::CacheExpire2(_) => (),
-        RollState::Done => {
-            actions.push(Action::UpdateDnskeyRrset);
-        }
-    }
-    actions
-}
-
-fn csk_roll(rollop: RollOp<'_>, ks: &mut KeySet) -> Result<(), Error> {
+fn zsk_double_signature_roll(
+    rollop: RollOp<'_>,
+    ks: &mut KeySet,
+) -> Result<(), Error> {
     match rollop {
         RollOp::Start(old, new) => {
-            // First check if the current CSK-roll state is idle. We need
+            // First check if the current ZSK-roll state is idle. We need
             // to check all conflicting key rolls as well. The way we check
             // is to allow specified non-conflicting rolls and consider
             // everything else as a conflict.
-            if let Some(rolltype) = ks.rollstates.keys().next() {
-                if *rolltype == RollType::CskRoll {
+            if let Some(rolltype) = ks.rollstates.keys().find(|k| {
+                **k != RollType::KskRoll && **k != RollType::KskDoubleDsRoll
+            }) {
+                if *rolltype == RollType::ZskDoubleSignatureRoll {
                     return Err(Error::WrongStateForRollOperation);
                 } else {
                     return Err(Error::ConflictingRollInProgress);
                 }
             }
             // Check if we can move the states of the keys
-            ks.update_csk(Mode::DryRun, old, new)?;
+            ks.update_zsk_double_signature(Mode::DryRun, old, new)?;
             // Move the states of the keys
-            ks.update_csk(Mode::ForReal, old, new)
-                .expect("Should have been check with DryRun");
+            ks.update_zsk_double_signature(Mode::ForReal, old, new)
+                .expect("Should have been checked with DryRun");
         }
         RollOp::Propagation1 => {
-            // Set the visiable time of new KSKs, ZSKs and CSKs to the current
+            // Set the visiable time of new ZSKs to the current time.
```

Comment:
There are other occurrences of this typo elsewhere too:
```suggestion
            // Set the visible time of new ZSKs to the current time.
```

Reply:
Done.